### ISSUE TITLE
Tcl verbatim strings can be code and should be syntax highlighted

### ISSUE THREAD

**Issue by d-led** [sublimehq/Packages#365]
a follow-up for SublimeTextIssues/DefaultPackages#131

curly braces in Tcl can be core, thus should not be highlighted as strings:

![image](https://cloud.githubusercontent.com/assets/1716584/15105747/86733ae6-15c4-11e6-9545-a8669cc6116e.png)

it should rather be:
![image](https://cloud.githubusercontent.com/assets/1716584/15105755/8e0b9b5e-15c4-11e6-9b9f-96ba70f275e0.png)

working example: http://ideone.com/7KsnMz

If it helps, [Np++](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/287ce9ec147d6d64224d8990f99385f57a240df4/scintilla/lexers/LexTCL.cxx) syntax for Tcl seems to work ok.

For another test case: square brackets inside quoted strings are [code](http://ideone.com/1kYRCX) too:

```tcl
set b 33

proc a {} {
	return "this is [set b 42] here, \[here it's $b\], highlighted too"
}

puts "[a] otherwise: $b"
```

**Comment by wbond** [comment]
Can you try out the [tcl-rewrite](https://github.com/sublimehq/Packages/tree/tcl-rewrite) branch and comment here with code samples of any errors you found?

**Comment by d-led** [comment]
@wbond ok! Will gradually do.

Seems to work ok on the first nesting level:

![image](https://cloud.githubusercontent.com/assets/1716584/15138547/e6e214d4-1690-11e6-9291-79d4ab93b5d8.png)

The syntax highlighting of the `proc c` body is probably not necessary, as it's anyway not good code. What seems to be missing in the above example: handling of semicolon in the nested code. 

---

I haven't looked at the syntax definition closer, thus, is `.sublime-syntax` "theoretically" expressive enough to handle unlimited recursion of Tcl syntax?

---

Other editors prefer to only highlight obvious identifiers or constants or leave out the quoted strings contents altogether. Github manages somehow:

```tcl
set b 33

proc a {} {
    return "this is [proc c {p1 p2 p3} { puts 42 }; set b 42;] here, \[here it's $b\], highlighted too"
}

puts "[a] otherwise: $b"
```

---

the original example works ok

![image](https://cloud.githubusercontent.com/assets/1716584/15138699/e9503fb0-1691-11e6-91af-6bd00d5bb49b.png)

**Comment by wbond** [comment]
The syntax definitions use the [`.sublime-syntax`](http://www.sublimetext.com/docs/3/syntax.html) format, which is different from the documentation you linked to. The documentation you linked to covers a third-party syntax that is converted to the old `.tmLanguage` syntax.

I just pushed up changes that fixed the `;` highlighted as being illegal.

The way that the `.sublime-syntax` is written, it would be extra work to try and exclude the embedded `[proc ...]` command substitution from the string. And from my relatively newly formed understand of Tcl syntax, that command substitution would be executed, correct?

You made the comment:

> Other editors prefer to only highlight obvious identifiers or constants or leave out the quoted strings contents altogether. Github manages somehow:

The code sample here on GitHub seems to highlight the string, the embedded `proc` in the string and `$b`. It seems the primary difference is that the rewrite I did for Sublime also handles character escapes, numeric literals inside of command substitutions and proc parameter highlighting. Since these (other than numeric literals) seem to be accurate with the Tcl language, I would presume they are improvements.
